tests/python/rest_api/test_tasks.py

```diff
@@ -5667,16 +5669,20 @@ def test_check_import_error_on_wrong_file_structure(self, tasks_with_shapes, for
             with open(self.tmp_dir / file, "w") as f:
                 f.write("Some text")
 
-        zip_file = zipfile.ZipFile(source_archive_path, mode="a")
-        for path in incorrect_files:
-            zip_file.write(self.tmp_dir / path, path)
+        with zipfile.ZipFile(source_archive_path, mode="a") as zip_file:
+            for path in incorrect_files:
+                zip_file.write(self.tmp_dir / path, path)
+
         task = self.client.tasks.retrieve(task_id)
 
         with pytest.raises(exceptions.ApiException) as capture:
             task.import_annotations(format_name, source_archive_path)
 
-            assert b"Check [format docs]" in capture.value.body
-            assert b"Dataset must contain a file:" in capture.value.body
+        error_message = str(capture.value)
```

Comment:
```suggestion
        error_message = capture.value.body.decode()
```

IMO, we should make sure that the error is in the place where we expect it to be.

Reply:
You are right, I'll fix it. I’ve changed the logic of the SDK `Client.wait_for_completion` method to raise a custom-defined exception instead of `ApiException`. It seemed incorrect to me that `ApiException` was raised with a 200 status code (but a 200 status code is the desired status in the requests API, even when a background job fails). That's why `.body` is gone.